snuba/datasets/events_format.py

```diff
@@ -47,20 +58,24 @@ def extract_extra_tags(
 
 
 def extract_nested(
-    nested_col: Mapping[str, Any], val_processor: Callable[[Any], TVal]
+    nested_col: Mapping[str, Any], val_processor: Callable[[Any], Optional[TVal]]
 ) -> Tuple[Sequence[str], Sequence[TVal]]:
     keys = []
     values = []
     for key, value in sorted(nested_col.items()):
         value = val_processor(value)
         if value is not None:
-            keys.append(_unicodify(key))
+            unicode_key = _unicodify(key)
```

Comment:
How certain are we keys cannot be None ?

Reply:
I think we can have confidence on the input type to unicodify by hunting down where extract_nested is used. Maybe is guaranteed to be a non Null string. Otherwise we should probably collect metrics first before asserting. If that fails the consumer fails.

Reply:
Pretty sure since
- The input has already gone through `as_dict_safe()` which strips out None keys
- We never handle that later so i'd expect to see clickhouse errors on insert if that was the case

Reply:
Any chance we can change the signature ?

Reply:
Signature of what? `unicodify`? It's used in a lot of places where the other types may be valid

Reply:
Sorry this comment was for the other case. I was talking about the signature of the extract_nested method. But the key is already str.